var/spack/repos/builtin/packages/axom/package.py

```diff
@@ -0,0 +1,497 @@
+    variant("raja",     default=True, description="Build with raja")
+    variant("umpire",   default=True, description="Build with umpire")
+
+    varmsg = "Build development tools (such as Sphinx, Uncrustify, etc...)"
+    variant("devtools",  default=False, description=varmsg)
+
+    # -----------------------------------------------------------------------
+    # Dependencies
+    # -----------------------------------------------------------------------
+    # Basics
+    depends_on("cmake@3.8.2:", type='build')
+    depends_on("cuda", when="+cuda")
+    depends_on("mpi", when="+mpi")
+
+    # Libraries
+    depends_on("conduit~shared+python", when="+python")
+    depends_on("conduit~shared~python", when="~python")
+    depends_on("conduit~shared+python+hdf5", when="+hdf5+python")
+    depends_on("conduit~shared+python~hdf5", when="~hdf5+python")
+    depends_on("conduit~shared~python+hdf5", when="+hdf5~python")
+    depends_on("conduit~shared~python~hdf5", when="~hdf5~python")
+
+    # HDF5 needs to be the same as Conduit's
+    depends_on("hdf5@1.8.19:1.8.999~mpi~cxx~shared~fortran", when="+hdf5")
+
+    depends_on("lua", when="+lua")
+
+    depends_on("scr", when="+scr")
+
+    depends_on("raja~openmp", when="+raja~openmp")
+    depends_on("raja+openmp", when="+raja+openmp")
+    depends_on("raja~openmp+cuda", when="+raja~openmp+cuda")
+    depends_on("raja+openmp+cuda", when="+raja+openmp+cuda")
+
+    depends_on("umpire~openmp", when="+umpire~openmp")
+    depends_on("umpire+openmp", when="+umpire+openmp")
+    depends_on("umpire~openmp+cuda", when="+umpire~openmp+cuda")
+    depends_on("umpire+openmp+cuda", when="+umpire+openmp+cuda")
+
+    depends_on("mfem~mpi~hypre~metis~gzstream", when="+mfem")
+
+    depends_on("python", when="+python")
+
+    # Devtools
+    depends_on("cppcheck", when="+devtools")
+    depends_on("doxygen", when="+devtools")
+    depends_on("graphviz", when="+devtools")
+    depends_on("python", when="+devtools")
+    depends_on("py-sphinx", when="+devtools")
+    depends_on("py-shroud", when="+devtools")
+    depends_on("uncrustify@0.61", when="+devtools")
+
+    def _get_sys_type(self, spec):
+        sys_type = spec.architecture
+        # if on llnl systems, we can use the SYS_TYPE
+        if "SYS_TYPE" in env:
+            sys_type = env["SYS_TYPE"]
+        return sys_type
+
+    def _get_host_config_path(self, spec):
+        hostname = socket.gethostname()
+        if "SYS_TYPE" in env:
+            # Are we on a LLNL system then strip node number
+            hostname = hostname.rstrip('1234567890')
+        filename = "{0}-{1}-{2}.cmake".format(hostname,
+                                              self._get_sys_type(spec),
+                                              spec.compiler)
+        dest_dir = self.stage.source_path
+        fullpath = os.path.abspath(pjoin(dest_dir, filename))
+        return fullpath
+
+    def hostconfig(self, spec, prefix):
```

Comment:
Can these variables not be passed through `cmake_args`?

Reply:
a generated host config is our source of provenance and also a way to config again outside of spack

Reply:
This is an example of how messy all of these options can be.  It would be a huge problem to add them to one command line. https://github.com/LLNL/axom/blob/develop/host-configs/rzansel-blueos_3_ppc64le_ib_p9-clang%408.0.1_nvcc_xlf.cmake